dev_guide/app_tutorials/ruby_on_rails.adoc

```diff
@@ -0,0 +1,344 @@
+
+----
+$ sudo yum install -y postgresql postgresql-server postgresql-devel
+----
+
+Next you need to initialize the database with:
+
+----
+$ sudo postgresql-setup initdb
+----
+
+This command will create the `/var/lib/pgsql/data` directory, in which the data will be stored.
+
+Start the database by typing:
+
+----
+$ sudo systemctl start  postgresql.service
+----
+
+When the database is running, create your `rails` user:
+
+----
+$ sudo -u postgres createuser -s rails
+----
+
+Note that the user we created is without password.
+
+== Write your application
+
+If you are starting your Rails application from scratch, you need to install the Rails gem at first.
+
+----
+$ gem install rails
+Successfully installed rails-4.2.0
+1 gem installed
+----
+
+After you install the Rails gem create a new application, with PostgreSQL as your database:
+
+----
+$ rails new rails-app --database=postgresql
+----
+
+Then move into your new application directory.
+
+----
+$ cd rails-app
+----
+
+In case you already have an application, make sure the `pg` (postgresql) gem is present in your `Gemfile`. If not
+edit your `Gemfile` by adding the gem:
+
+----
+gem 'pg'
+----
+
+To generate a new `Gemfile.lock` with all your dependencies run:
+
+----
+$ bundle install
+----
+
+In addition to using the `postgresql` database with the `pg` gem, you'll also need to ensure the `config/database.yml` is using the `postgresql` adapter.
+
+Make sure you updated `default` section in the `config/database.yml` file, so it looks like this:
+
+----
+default: &default
+  adapter: postgresql
+  encoding: unicode
+  pool: 5
+  host: localhost
+  username: rails
+  password:
+----
+
+Create your application's development and test databases by using this rake command:
+
+----
+$ rake db:create
+----
+
+This will create `development` and `test` database in your PostgreSQL server.
+
+=== Welcome page
+
+Since Rails 4 no longer serves a static `public/index.html` page in production, we need to create a 
+new root page.
+
+In order to have a custom welcome page we need to do following steps:
+
+* Create a *controller* with an index action 
+* Create a *view* page for the `welcome` controller `index` action
+* Create a *route* that will serve applications root page with the created *controller* and *view*  
+
+Rails offers a generator that will do all this necessary steps for you. Just type:
+
+----
+$ rails generate controller welcome index
+----
+
+All the necessary files have been created, now we just need to edit line 2 in `config/routes.rb`
+file to look like:
+----
+root 'welcome#index'
+---- 
+
+Run the rails server to verify the page is available
+
+----
+$ rails server
+----
+
+You should see your page by visiting link:http://localhost:3000[http://localhost:3000] in our browser.
+In case you don't see the page, check the logs that are output to your server to debug.
+
```

Comment:
you're transitioning from "writing/testing the local app" to "moving the app into openshift", so i think it would be good to either have a paragraph explaining that, or have a separate section heading "Configuring the Application For OpenShift" that these topics are nested under.